File: backend/internal/domain/ports/task_repository.go
package ports

import (
	"context"

	"todolist/backend/internal/domain/task"
)

type TaskRepository interface {
	FindByID(ctx context.Context, id string) (*task.Aggregate, error)
	FindByTodoListID(ctx context.Context, todoListID string) ([]*task.Aggregate, error)
	FindByProjectID(ctx context.Context, projectID string) ([]*task.Aggregate, error)
	Save(ctx context.Context, aggregate *task.Aggregate) error
}

File: frontend/src/dominio/puertos/TaskRepository.ts
import type { TaskAggregate } from '../entidades/TaskAggregate'

export interface TaskRepository {
  findById(id: string): TaskAggregate | null
  findByTodoListId(todoListId: string): TaskAggregate[]
  findByProjectId(projectId: string): TaskAggregate[]
  save(task: TaskAggregate): void
}
